thrift/src/main/java/com/linecorp/armeria/internal/thrift/ThriftServiceMetadata.java

```diff
@@ -56,8 +60,21 @@
      * interfaces.
      */
     public ThriftServiceMetadata(Object implementation) {
-        requireNonNull(implementation, "implementation");
-        interfaces = init(implementation);
+        this(ImmutableList.of(implementation));
+    }
+
+    /**
+     * Creates a new instance from a list of Thrift service implementations, while each service can implement
+     * one or more Thrift service interfaces.
+     */
+    public ThriftServiceMetadata(Iterable<?> implementations) {
+        requireNonNull(implementations, "implementations");
+
+        final Builder<Class<?>> interfaceBuilder = ImmutableSet.builder();
```

Comment:
Could you use `ImmutableSet.Builder` because if we need `ImmutableList.Builder` this static import causes conflicts :-)
```suggestion
        final ImmutableSet.Builder<Class<?>> interfaceBuilder = ImmutableSet.builder();
```